### PR TITLE
Added Circular_Queue

### LINKED ISSUE #1264: Circular queue in C, C++

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1264

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Circular_Queue in c and c++

#### Languages Used:
- C
- C++

#### Files Added:
- Circular_Queue.c
- Circular_Queue.cpp

### CODE REVIEW

File: Circular_Queue/Circular_Queue.c [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    printf("3.Display\n");
+    
+    do
+    {
+        printf("Enter choice: ");
+        scanf("%d",&choice);
+        switch(choice)
+        {
+            case 1: 
+                    printf("Enter the value: ");
```
Comment (marwahmanbir) [inline Circular_Queue/Circular_Queue.c]: There should be a 16-space indentation for each case statement until line 38.

File: Circular_Queue/Circular_Queue.c [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+/*   
+Deletion:
+1) Check whether queue is Empty means check (front==-1).
+2) If it is empty then display Queue is empty. If queue is not empty then Check if (front==rear) 
+   if it is true then set front=rear= -1 else check if (front==size-1), if it is true then set 
+   front=0 and return the element.
+*/
+// Deleting from queue
+void deletequeue()
```
Comment (marwahmanbir) [inline Circular_Queue/Circular_Queue.c]: Use camelCase here (``deleteQueue``).

File: Circular_Queue/Circular_Queue.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+int front = -1, rear = -1, size = 5;
+
+void insert (int value);
+void deletequeue();
+void display();
+
+int main()
+{
+    int choice, value;
+    cout << "1.Insert\n";
```
Comment (marwahmanbir) [inline Circular_Queue/Circular_Queue.cpp]: Please remove spaces around ``<<`` and ``>>`` in ``cout`` and ``cin`` statements, they are unnecessary and the compiler simply ignores them.

File: Circular_Queue/Circular_Queue.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    cout << "3.Display\n";
+    
+    do
+    {
+        cout << "Enter choice: ";
+        cin >> choice ;
+        switch(choice)
+        {
+            case 1: 
+                    cout << "Enter the value: ";
```
Comment (marwahmanbir) [inline Circular_Queue/Circular_Queue.cpp]: Fix indentation (16-space) in all case statements until line 39.

File: Circular_Queue/Circular_Queue.c [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        return ;
+    }
+    printf("Element deleted from queue is : %d\n", queue[front]);
+
+    if (front == rear)
+    {
+        front = -1;
+        rear = -1;
+    }
+    else
```
Comment (marwahmanbir) [inline Circular_Queue/Circular_Queue.c]: In the previous if-else block, you have added a blank line before each else statement, but not in deletequeue and display method. Please do some unification.

File: Circular_Queue/Circular_Queue.cpp [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        return ;
+    }
+    cout << "Element deleted from queue is : " << queue[front] << "\n";
+
+    if (front == rear)
+    {
+        front = -1;
+        rear = -1;
+    }
+    else
```
Comment (marwahmanbir) [inline Circular_Queue/Circular_Queue.cpp]: In the previous if-else block, you have added a blank line before each else statement, but not in deletequeue and display method. Please do some unification.

**Comment by somya-kapoor** [comment]
@marwahmanbir Done please review

**Comment by somya-kapoor** [comment]
@MastersAbh Done please review.

**Comment by somya-kapoor** [comment]
@jainaman224 Done, Please review.

File: Circular_Queue/Circular_Queue.c [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        printf("\nQueue is Empty"); 
+        return; 
+    } 
+    
+    printf("\nCircular Queue elements are: ");
+    
+    if (front <= rear) 
+    { 
+        for (int i = front; i <= rear; i++) 
+            printf("%d ",queue[i]);
```
Comment (MastersAbh) [inline Circular_Queue/Circular_Queue.c]: space after comma

File: Circular_Queue/Circular_Queue.cpp [hunk 8]
```diff
@@ -0,0 +1,151 @@
+/* 
+Circular Queue is based on FIFO (First In First Out) principle and the last position 
+is connected back to the first position to make a circle.
+*/
+
+#include <iostream>
```
Comment (MastersAbh) [inline Circular_Queue/Circular_Queue.cpp]: remove space before <

File: Circular_Queue/Circular_Queue.c [hunk 9]
```diff
@@ -0,0 +1,150 @@
+/* 
+Circular Queue is based on FIFO (First In First Out) principle and the last position 
+is connected back to the first position to make a circle.
+*/
+
+#include <stdio.h>
```
Comment (MastersAbh) [inline Circular_Queue/Circular_Queue.c]: remove space before <

**Comment by somya-kapoor** [comment]
@MastersAbh Done, Please review.

**Comment by somya-kapoor** [comment]
@marwahmanbir Please review

**Comment by somya-kapoor** [comment]
@jainaman224 @marwahmanbir Please review

**Comment by somya-kapoor** [comment]
@MastersAbh I ran with the same input but it is running fine.
Input :
1
1
1
2
1
3
1
4
2
1
3
3 // For showing the queue.

**Comment by jainaman224** [comment]
@MastersAbh Please have a look. And if possible add repl.it link.

**Comment by MastersAbh** [comment]
Hi, I found out the issue.
The problem is that every time you type 3 as your choice, the program displays and exits and therefore, no further inputs are allowed.
giving 3 as an input serves two purposes : displaying and exiting. @somya-kapoor , can 4 or some dummy number be used for exiting the program, else it would be fine if @jainaman224 suggests so.

**Comment by somya-kapoor** [comment]
@jainaman224  Please review

**Pull request merged**